instrumentation/pekko/pekko-actor-1.0/javaagent/src/test/scala/io/opentelemetry/javaagent/instrumentation/pekkoactor/v1_0/PekkoSchedulerTest.scala

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.pekkoactor.v1_0
+
+import io.opentelemetry.api.GlobalOpenTelemetry
+import io.opentelemetry.api.trace.Span
+import org.apache.pekko.actor.ActorSystem
+import org.apache.pekko.pattern.after
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import scala.concurrent.{Await, Future}
+import scala.concurrent.duration.DurationInt
+
+class PekkoSchedulerTest {
+
+  @Test
+  def checkThatSpanWorksWithPekkoScheduledEvents(): Unit = {
+    val system = ActorSystem("my-system")
+    implicit val executionContext = system.dispatcher
+    val tracer = GlobalOpenTelemetry.get.getTracer("test-tracer")
+    val initialSpan = tracer.spanBuilder("test").startSpan
+    try {
+      val scope = initialSpan.makeCurrent()
+      val futureResult = for {
+        result1 <- Future {
+          assertThat(Span.current().getSpanContext().getTraceId())
+            .isEqualTo(initialSpan.getSpanContext().getTraceId())
+          assertThat(Span.current().getSpanContext().getSpanId())
+            .isEqualTo(initialSpan.getSpanContext().getSpanId())
```

Comment:
perhaps extract this to separate method as this assertion block is repeated 3 times